src/gluonts/model/rotbaum/_model.py

```diff
@@ -196,6 +201,15 @@ def fit(
         del cell_values_dict
         gc.collect()
 
+        df = pd.DataFrame({"preds": self.sorted_train_preds})
+        df["bins"] = df["preds"].apply(lambda x: self.preds_to_id[x])
+        final_id = df["bins"].drop_duplicates().values[-1]
```

Comment:
`bins = df["bins"].drop_duplicates().values()`, so you only have to compute once and then take the last and second to last element